app/api/lectures.js

```diff
@@ -0,0 +1,276 @@
+        const lectures = await db.Lecture.findAll({
+            where: { courseId: courseId }
+        })
+        if (lectures == []) {   // if no lectures are in this course
+            res.status(204).send()
+        }
+        else {
+            res.status(200).json({
+                "lecture": lectures
+            })
+        }
+    }
+})
+
+router.post('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    var lecture     // will hold the returned lecture object from database
+
+    if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+    }
+    else if (enrollment.role == 'teacher') {
+        // create lecture object
+        let newLec = req.body
+        newLec['courseId'] = courseId
+        try {
+            lecture = await db.Lecture.create(newLec)
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to create lecture object"})
+            return
+        }
+
+        // create lecture-section association for all sections in course
+        const sectionIds = await getSectionsIdsFromCourse(courseId);
+        try {
+            // iterate through each section in this course and add relationship
+            for (let i = 0; i < sectionIds.length; i++) {
+                await db.LectureForSection.create({
+                    lectureId: lecture.id,
+                    sectionId: sectionIds[i]
+                    // TODO: no other fields are being added here
+                    // ...should it be added here, or in /:section_id/lectures ?
+                })
+            }
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to create association between lecture & this course' sections"})
+            return
+        }
+        res.status(201).json(lecture)   // all good, return lecture object
+    }
+    else {      // if user is not a teacher
+        res.status(403).send()
+    }
+})
+
+router.put('/:lecture_id', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const lectureId = req.params.lecture_id
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    const lecture = await getLecture(lectureId)
+    
+    if (lecture == null) {  // if passed in lectureId does not exist
+        res.status(404).send({error: "Lecture of this id does not exist"})
+        return
+    }
+    else if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+    }
+    else if (enrollment.role == 'teacher') {
+        try {
+            await db.Lecture.update(
```

Comment:
Before you put the body directly into the update, we have to make sure something that the user isn't allowed to update is in the body. For example, what if the teacher had the courseId in the body? I'd recommend creating a lectures service like Mitch and I have done to create these validation methods. They're really simple